### ISSUE TITLE
Support rancher behind an ELB in AWS with SSL

### ISSUE THREAD

**Issue by cjellick** [rancher/rancher#1812]
We support running rancher in HA mode behind a number of different proxies. However, our current implementation does not support ELB. This is because Rancher uses websockets and ELB, when in HTTP/HTTPS mode, does not support websockets.

ELB does support websockets if it is configured in TCP mode and [Proxy Protocol](http://www.haproxy.org/download/1.5/doc/proxy-protocol.txt) is explicitly enabled. But we had to make some changes on the rancher side to support Proxy Protocol and this issue represents that work.

So now, if you configure your ELB to be in Proxy Protocol mode, Rancher will work with no additional configuration.
High level steps to test (assuming SSL):
1. Get a certificate for your domain (For example, I use craig.rancher.io). I can help you with this step.
2. Create an instance in amazon and run rancher/server on it.
3. In the AWS UI, create an ELB and configure it for TCP mode.*
4. From the AWS CLI, configure the above ELB to use Proxy protocol*
5. Verify basic functionality of rancher works: deploying a container, viewing stats, logs, exec.

\* Detailed steps for setting up a Proxy Protocol ELB: http://docs.aws.amazon.com/ElasticLoadBalancing/latest/DeveloperGuide/enable-proxy-protocol.html
\* Your ELB should be configured with TLS/SSL for the frontend and TCP for the backend. This causes SSL to be terminated at the load balancer.

**Comment by johnrengelman** [comment]
@cjellick I'm trying to out with v0.33.0-rc3 and I can't load the main page at all. Just get a 400 error.
If I switch the ELB back to HTTPS/HTTP then I get the page but I get an error about the websockets not connecting. Am I missing some other configuration that I need to pass to the server when running in the SSL/TCP mode on the ELB?

**Comment by deniseschannon** [comment]
I'm pretty sure that v0.33.0-rc3 does not have this fix, but the next RC will. There was a bug that was found after the RC2 was cut and we reverted the websocket proxy for RC3.

**Comment by johnrengelman** [comment]
Ah. Ok. That makes sense then. I didn't see the revert in the commit history. Thanks.

**Comment by aruneli** [comment]
Verified Rancher behind ELB configured to be in proxy protocol mode. It should work starting v0.34.0.

**Comment by johnrengelman** [comment]
Now the page loads up, but I get a mixed content error in Chrome that prevents the site from loading:
```
Mixed Content: The page at 'https://rancher.connectedfleet.io/static/' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://rancher.connectedfleet.io/v1/projects?limit=1000'. This request has been blocked; the content must be served over HTTPS.
```

**Comment by aruneli** [comment]
@johnrengelman This will be fixed in #1876
